pkg/sql/opt/memo/typing.go

```diff
@@ -110,6 +110,29 @@ func AggregateOverloadExists(agg opt.Operator, typ *types.T) bool {
 	return false
 }
 
+// FindFunction returns the function properties and overload of the function
+// with the given name and argument types matching the given inputs.
+func FindFunction(
```

Comment:
This and findOverload only differ in their implementation by how they get the `i`th argument type. It may be worthwhile to merge them into one so we have slightly less code duplication.